iceoryx_hoofs/include/iceoryx_hoofs/log/platform_building_blocks/logger.hpp

```diff
@@ -63,10 +72,23 @@ class Logger : public BaseLogger
 
     ~Logger() = default;
 
+    /// @brief Access to the logger singleton instance
+    /// @return a reference to the active logger
     static Logger& get() noexcept;
 
+    /// @brief Initializes the logger
+    /// @param[in] logLevel is the log level which will be used to determine which messages will be logged. By default
+    /// it is everything with a log level higher than specified by the `IOX_LOG_LEVEL` environment variable or equal to
+    /// INFO if the environment variable is not set.
+    /// @note The function uses 'getenv' which is not thread safe and can result in undefined behavior when it is called
```

Comment:
Actually it would make sense that you do not use the `getenv` call directly but you forward it to the `iceoryx_platform`. The reason is that some platforms have a thread-safe implementation and we could use them if they are available.

See: `secure_getenv` for instance on linux. Furthermore, we would be able to ensure it's thread-safety (when required) directly on the platform level with a static local mutex inside our `iox_getenv` and `iox_setenv` functions (something for the future). We still would have the problem with outside change maybe.

Reply:
@elfenpiff this was supposed to be a layer below the platform and therefore could not be used. But it seems this will change and then it would be possible. I would add an action item to the issue to redo the layering for the logger and use the platform directly.

Reply:
@elfenpiff added the task to the issue